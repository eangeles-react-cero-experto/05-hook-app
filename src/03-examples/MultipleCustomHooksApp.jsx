import { useFetch } from "../hooks/useFetch";

export const MultipleCustomHooksApp = () => {
  const { data, isLoading, error } = useFetch(
    "https://api.breakingbadquotes.xyz/v1/quotes"
  );

  const { quote, author } = !!data && data[0];

  return (
    <>
      <h1>BreakinBad Quotes</h1>
      <hr />

      {isLoading ? 123 : 123}

      <div className="alert alert-info text-center">Loading...</div>

      <blockquote className="blockquote text-end">
        <p>{quote}</p>
        <footer className="blockquote-footer">{author}</footer>
      </blockquote>

      <br />

      <button className="btn btn-primary">Next Quote</button>
    </>
  );
};
