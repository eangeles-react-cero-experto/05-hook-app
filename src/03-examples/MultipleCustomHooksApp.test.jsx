import { fireEvent, render, screen } from "@testing-library/react";
import { MultipleCustomHooksApp } from "./MultipleCustomHooksApp.jsx";
import { useCounter, useFetch } from "../hooks/index.js";

jest.mock("../hooks/useFetch");
jest.mock("../hooks/useCounter");

describe("Test in MultipleCustomHooksApp", () => {
  const mockIncrement = jest.fn();
  useCounter.mockReturnValue({
    counter: 1,
    incrementCounter: mockIncrement,
  });

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it("should match with snapshot", () => {
    useFetch.mockReturnValue({ data: null, isLoading: true, error: null });

    const { container } = render(<MultipleCustomHooksApp />);
    expect(container).toMatchSnapshot();
  });

  it("should show the default component", () => {
    render(<MultipleCustomHooksApp />);

    expect(screen.getByText("BreakingBad Quotes")).toBeTruthy();
    expect(screen.getByText("Loading...")).toBeTruthy();
    expect(screen.getByRole("button", { name: "Next Quote" })).toBeTruthy();
  });

  it("should show a quote", () => {
    useFetch.mockReturnValue({
      data: [
        {
          author: "John Due",
          quote: "Hello World",
        },
      ],
      isLoading: false,
      error: null,
    });

    render(<MultipleCustomHooksApp />);
    expect(screen.getByText("John Due")).toBeTruthy();
    expect(screen.getByText("Hello World")).toBeTruthy();

    const btnNextQuote = screen.getByRole("button", { name: "Next Quote" });
    expect(btnNextQuote.disabled).toEqual(false);
  });

  it("should call the increment function", () => {
    render(<MultipleCustomHooksApp />);
    const btnNextQuote = screen.getByRole("button", { name: "Next Quote" });
    fireEvent.click(btnNextQuote);
    expect(mockIncrement).toHaveBeenCalled();
  });
});
