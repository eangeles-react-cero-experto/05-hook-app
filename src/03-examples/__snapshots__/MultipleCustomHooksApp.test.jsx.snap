// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in MultipleCustomHooksApp should match with snapshot 1`] = `
<div>
  <h1>
    BreakingBad Quotes
  </h1>
  <hr />
  <div
    class="alert alert-info text-center"
  >
    Loading...
  </div>
  <br />
  <button
    class="btn btn-primary"
    disabled=""
  >
    Next Quote
  </button>
</div>
`;

exports[`Test in MultipleCustomHooksApp should match with snapshott 1`] = `
<div>
  <h1>
    BreakingBad Quotes
  </h1>
  <hr />
  <div
    class="alert alert-info text-center"
  >
    Loading...
  </div>
  <br />
  <button
    class="btn btn-primary"
    disabled=""
  >
    Next Quote
  </button>
</div>
`;
